components/layout_2020/table/layout.rs

```diff
@@ -245,9 +245,20 @@ impl<'a> TableLayout<'a> {
         layout_context: &LayoutContext,
         writing_mode: WritingMode,
     ) {
+        // It's not clear whether `inline-size: stretch` allows fixed table mode or not,
+        // we align with Gecko and Blink.
+        // <https://github.com/w3c/csswg-drafts/issues/10937>.
         let is_in_fixed_mode = self.table.style.get_table().clone_table_layout() ==
             TableLayoutMode::Fixed &&
-            !self.table.style.box_size(writing_mode).inline.is_auto();
+            match self.table.style.box_size(writing_mode).inline {
+                Size::Numeric(_) => true,
+                Size::Keyword(keyword) => match keyword {
+                    SizeKeyword::Initial | SizeKeyword::MaxContent => false,
+                    SizeKeyword::MinContent | SizeKeyword::FitContent | SizeKeyword::Stretch => {
+                        true
+                    },
+                },
+            };
```

Comment:
Could this be a `matches!`?

Something like this:
```suggestion
            matches!(
                self.table.style.box_size(writing_mode).inline,
                Size::Numeric(_) |
                Size::Keyword(SizeKeyword::MinContent) |
                Size::Keyword(SizeKeyword::FitContent) |
                Size::Keyword(SizeKeyword::Stretch)
             );
```

Reply:
This wouldn't cover the `fit-content()` function when we add it. So to align with Gecko & Blink, I'm going with

```rs
!matches!(
    self.table.style.box_size(writing_mode).inline,
    Size::Keyword(SizeKeyword::Initial | SizeKeyword::MaxContent)
);
```

- https://searchfox.org/mozilla-central/rev/1e8cec3727d6e09f4af41bb3d202b7a4c326ed84/layout/tables/nsTableFrame.cpp#3482
- https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/style/computed_style.h;drc=572789ee61c7e857aa288a7ac7768b2b0b4f4dcc;l=1924